packages/expo-updates/src/Updates.types.ts

```diff
@@ -71,6 +79,10 @@ export type UpdateCheckResultFailure = {
    * Whether a roll back to embedded update is available.
    */
   isRollBackToEmbedded: false;
+  /**
+   * If no new update was found, this contains an optional message with additional info.
+   */
+  message?: string;
```

Comment:
My gut says that we should use an enum here since consumers of the API may wish to do something programmatically with the result, and we want to reserve the right to do things like translate, change the exact wording, etc. Something like `reason: UpdateCheckResultFailureReason` and the following:

```
enum UpdateCheckResultFailureReason {
  /**
   * A update manifest was received from the update server, but the update has been previously launched on this device and never successfully launched.
   */
  PREVIOUSLY_FAILED_UPDATE_RECEIVED = 'previouslyFailedUpdateReceived',
  /**
   * A update manifest was received from the update server, but the update is not launchable, or does not pass the configured selection policy
   */
  DOES_NOT_PASS_SELECTION_POLICY = 'doesNotPassSelectionPolicy',
  ...
}
```

Then use enums in the native layer as well.

Reply:
Yep! That's by design. We should be explicit about adding new reasons so that we consider all cases.

Reply:
Ok reason enum is added